sensor/kubernetes/listener/resources/selector.go

```diff
@@ -1,45 +1,118 @@
 package resources
 
 import (
+	"math"
+
 	"k8s.io/apimachinery/pkg/labels"
 )
 
-// selector is a restricted version of labels.Selector
+// labelWithLen is label.Labels with added Len() function
+type labelsWithLen interface {
+	Has(label string) (exists bool)
+	Get(label string) (value string)
+	Len() uint
+}
+
+// selector is a restricted version of selectorWrapper
 type selector interface {
+	Matches(labelsWithLen) bool
+}
+
+// internalSelector is a restricted version of labels.Selector
+type internalSelector interface {
 	Matches(labels.Labels) bool
 }
 
+type labelWrapper struct {
+	labels    labels.Labels
+	numLabels uint
```

Comment:
I think, it would be enough to store a reference to the `map[string]string` here.

Reply:
Wouldn't we lose performance by having to create a labels.labels object every time we call `Matches(labels labels.labels)` in this case, or is that negligible?